### PR TITLE
Add tooling to fix KB index data

### PR DESCRIPTION
**Pull request by lferran**
### Description

- Added a way to recompute only the vectors of resources in a KB.
- Added script to fix production's indexes. Given a KB, it will:
   - clean it's data from the node.
   - upgrade the shard to the latest version
   - Then send all resources to recompute again

### How was this PR tested?
I tested it manually

### CODE REVIEW

File: nucliadb_client/nucliadb_client/resource.py [hunk 1]
```diff
@@ -69,15 +70,33 @@ def __init__(
...
+        field_type = field_type or []
+        if field_type:
+            params["field_type"] = field_type
+
+        if extracted:
+            if "extracted" not in show:
+                show.append("extracted")
+            params["extracted"] = extracted
+
+        response = self.http_reader_v1.get("?", params=params, timeout=timeout).content
```
Comment (jotare) [inline nucliadb_client/nucliadb_client/resource.py:95]: Is `?` needed when passing `params`?

File: nucliadb_client/nucliadb_client/resource.py [hunk 2]
```diff
@@ -69,15 +70,33 @@ def __init__(
...
         return NucliaDBResource.parse_raw(response)
 
     def update(self):
-        response = self.http_writer_v1.get("").content
+        response = self.http_reader_v1.get("").content
```
Comment (jotare) [inline nucliadb_client/nucliadb_client/resource.py:99]: What's the point to get the resource on `update` function?

**Review by jotare — CHANGES_REQUESTED** [review]

File: nucliadb_client/examples/kb_admin.py [hunk 3]
```diff
@@ -19,91 +19,231 @@
...
+
+from nucliadb.models.common import FieldTypeName
+from nucliadb.reader.api.models import FIELD_NAMES_TO_PB_TYPE_MAP
 from nucliadb_client.client import NucliaDBClient
+from nucliadb_client.resource import Resource
+
+VECTORS_DIMENSION = 128
+DEFAULT_INGEST_GRPC_PORT = 8030
+DEFAULT_API_HTTP_PORT = 8080
+DEFAULT_TRAIN_GRPC_PORT = 8080
```
[the remark this answers is not in the document]
Comment (lferran) [reply nucliadb_client/examples/kb_admin.py:43]: train doesn't expose a HTTP service right now. It only exposes the grpc servicer, which happens to serve at port 8080

**Review by jotare — APPROVED** [review]

File: nucliadb_client/examples/fix-prod-index-data.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    #     grpc=INGEST_GRPC_PORT,
+    #     http=API_HTTP_PORT,
+    #     train_port=TRAIN_GRPC_PORT,
+    #     reader_host=INTERNAL_SERVICES["reader"],
+    #     writer_host=INTERNAL_SERVICES["writer"],
+    #     search_host=INTERNAL_SERVICES["search"],
+    #     train_host=INTERNAL_SERVICES["train"],
+    #     grpc_host=INTERNAL_SERVICES["ingest"],
+    #     dry_run=args.dry_run,
+    # )
```
Comment (jotare) [inline nucliadb_client/examples/fix-prod-index-data.py:65]: To delete?

**Pull request merged**